### PR TITLE
It's finally ready for a pr :D

### PR DESCRIPTION
**Pull request by amenyxia**
I rewrote the entire structure of everything to be entirely modular
Switched input to keyboard instead of controller
Added stopping keys
And so much more

I also re-made your pre-existing strategy into the format I created
This should run better and be easier to make changes to
I will continue to make new modules and stuff afterwards, but this is just to get a foundation merged so that it's properly setup incase anyone else wants to contribute

### CODE REVIEW

File: main/base_classes.py
Comment (Junkgineer) [inline main/base_classes.py:1]: As someone who is overly addicted to classes and modules, I for one like this approach. Edit: I'm just some guy so take my comments with a grain of salt!

File: thought_process.txt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Basically I am going to write this while I do shit so that all the changes I'm doing make sense :D
+
+Something I notice beforehand is you're using threading
+This doesn't really do anything other than hurt performance because you could run the image stuff in the same thread
+Due to the GIL (global interpreter lock) only one thread can actually use cpu resources at a time, meaning that it'd probably do you just as much good to run everything together
+
+First of all, I'll make the main just a py file because I'm not used to working with jupyter notebooks
```
Comment (Junkgineer) [inline thought_process.txt]: Thank you for this!

**Comment by thrailkillr** [comment]
doesnt work on linux since DXCam is windows only

**Comment by amenyxia** [comment]
> doesnt work on linux since DXCam is windows only

I don't use Linux myself so I don't have knowledge of what would work well. If you know any high performance screen capturing libraries for Linux I'd be more than happy to add that in, but I don't have a Linux machine with ultrakill on it to test. Alternatively, if the PR is accepted you could just create your own PR that integrates a second library for Linux specific functionality.
Also that isn't the biggest issue for Linux, the input system I have coded currently uses windows specific functionality in order to get the game to properly register the inputs, and once again I don't know what would work with Linux, so if you or someone else wants to help with a secondary input system for Linux that would also be appreciated.

**Comment by marmust** [comment]
Damn thanks alot :) sorry i wasnt available to merge it earlier, this is a major improvement :D

**Pull request merged**